### PR TITLE
Add Trino 460 release notes

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Assemble the release notes for the upcoming Trino 460 release.

Also see https://github.com/trinodb/trino/pulls?q=is%3Apr+is%3Aclosed+milestone%3A460

## Release notes

- [x] This is not user-visible or is docs only, and no release notes are required.

## Verification for each pull request

Format: PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 26 Sep 2024

* #23577 ✅ rn ✅ docs
* #23558 ✅ rn ✅ docs
* #23574 ✅ rn ✅ docs
* #22728 ✅ rn ✅ docs
* #23569 ✅ rn ✅ docs
* #23579 ✅ rn ✅ docs
* #23580 ✅ rn ✅ docs
* #23584 ✅ rn ✅ docs
* #23587 ✅ rn ✅ docs
* #23550 ✅ rn ✅ docs
* #23578 ✅ rn ✅ docs

## 27 Sep 2024

* #23429 ✅ rn ✅ docs
* #23483 ✅ rn ✅ docs
* #23581 ✅ rn ✅ docs
* #23507 ✅ rn ✅ docs
* #21603 ✅ rn ✅ docs
* #23269 ✅ rn ✅ docs
* #23201 ✅ rn ✅ docs
* #23597 ✅ rn ✅ docs

## 28 Sep 2024

* #23553 ✅ rn ✅ docs

## 29 Sep 2024

* #23599 ✅ rn ✅ docs
* #23595 ✅ rn ✅ docs
* #23602 ✅ rn ✅ docs

## 30 Sep 2024

* #23586 ✅ rn ✅ docs
* #23561 ✅ rn ✅ docs
* #23605 ✅ rn ✅ docs
* #23593 ✅ rn ✅ docs
* #23607 ✅ rn ✅ docs
* #23609 ✅ rn ✅ docs
* #23604 ✅ rn ✅ docs
* #23615 ✅ rn ✅ docs
* #23616 ✅ rn ✅ docs
* #23554 ✅ rn ✅ docs
* #23618 ✅ rn ✅ docs
* #23612 ✅ rn ✅ docs
* #23623 ✅ rn ✅ docs

## 1 Oct 2024

* #23627 ✅ rn ✅ docs
* #21700 ✅ rn ✅ docs
* #23625 ✅ rn ✅ docs
* #23629 ✅ rn ✅ docs
* #23630 ✅ rn ✅ docs
* #23591 ✅ rn ✅ docs
* #23633 ✅ rn ✅ docs
* #23614 ✅ rn ✅ docs
* #23588 ✅ rn ✅ docs
* #23637 ✅ rn ✅ docs
* #23636 ✅ rn ✅ docs

## 2 Oct 2024

* #23644 ✅ rn ✅ docs
* #23647 ✅ rn ✅ docs
* #23646 ✅ rn ✅ docs
* #23548 ✅ rn ✅ docs
* #23635 ✅ rn ✅ docs
* #23650 ✅ rn ✅ docs
* #21236 ✅ rn ✅ docs
* #22248 ✅ rn ✅ docs

## 3 Oct 2024

* #23589 ✅ rn ✅ docs
* #23653 ✅ rn ✅ docs
* #23651 ✅ rn ✅ docs
* #23649 ✅ rn ✅ docs
* #23352 ✅ rn ✅ docs
* #23659 ✅ rn ✅ docs
* #23656 ✅ rn ✅ docs
* #23658 ✅ rn ✅ docs
* #23660 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-460.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+* Fix writing of bloom filter columns in Parquet files. ({issue}`22701`)
+
+## Iceberg connector
+
+* Add support for using an [Alluxio cluster as file system
+  cache](/object-storage/file-system-alluxio). ({issue}`21603`)
+* Ensure table columns are cached in Glue even when table comment is too long.
+  ({issue}`23483`)
+* Fix queries stuck in the planning phase for a long time when there are large
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-460.md:36]: @martint @raunaqmorarka ?

File: docs/src/main/sphinx/release/release-460.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+## Hive connector
+
+* Add support for using an [Alluxio cluster as file system
+  cache](/object-storage/file-system-alluxio). ({issue}`21603`)
+* Fix writing of bloom filter columns in Parquet files. ({issue}`22701`)
+
+## Hudi connector
+
+* Fix writing of bloom filter columns in Parquet files. ({issue}`22701`)
```
Comment (raunaqmorarka) [inline docs/src/main/sphinx/release/release-460.md:28]: i forgot we don't have writes in hudi, please skip this one

File: docs/src/main/sphinx/release/release-460.md (hunk 1 above)
Comment (raunaqmorarka) [reply docs/src/main/sphinx/release/release-460.md:36]: Maybe:
Reduce planning time for queries on tables with a large number of structural type fields

Comment (mosabua) [reply docs/src/main/sphinx/release/release-460.md]: Done

File: docs/src/main/sphinx/release/release-460.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-460.md]: Haha.. me too. Its time we get that update to Hudi 1.0 and the related improvements and more features ;-)

File: docs/src/main/sphinx/release/release-460.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Release 460 (2 Oct 2024)
+
+## Atop connector
+
+* {{breaking}} Remove the Atop connector. ({issue}`23550`)
+
+## ClickHouse connector
+
+* Improve performance of listing columns. ({issue}`23429`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-460.md:9]: In what context? Which queries does it affect? Queries on `information_schema.columns`?

File: docs/src/main/sphinx/release/release-460.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+* Improve performance for queries comparing `varchar` columns. ({issue}`23558`) 
+* Improve performance for queries using `varchar` columns for `IN` comparisons. ({issue}`23581`)
+* Improve performance for queries with complex expression for `LIKE` clauses. ({issue}`23591`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-460.md:12]: Improve performance for queries with complex expressions involving `LIKE`.

File: docs/src/main/sphinx/release/release-460.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+## Delta Lake connector
+
+* Add support for using an [Alluxio cluster as file system
+  cache](/object-storage/file-system-alluxio). ({issue}`21603`)
+* Add support for WASBS to [](/object-storage/file-system-azure). ({issue}`23548`)
+* Fix writing of bloom filter columns in Parquet files. ({issue}`22701`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-460.md:19]: What's fixed? How was it broken?

File: docs/src/main/sphinx/release/release-460.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+* Add support for WASBS to [](/object-storage/file-system-azure). ({issue}`23548`)
+
+## Iceberg connector
+
+* Add support for using an [Alluxio cluster as file system
+  cache](/object-storage/file-system-alluxio). ({issue}`21603`)
+* Add support for WASBS to [](/object-storage/file-system-azure). ({issue}`23548`)
+* Ensure table columns are cached in Glue even when table comment is too long. ({issue}`23483`)
+* Reduce planning time for queries on tables with a large number of structural
```
Comment (martint) [inline docs/src/main/sphinx/release/release-460.md:38]: Reduce planning time for queries on columns containing a large number of nested fields.

File: docs/src/main/sphinx/release/release-460.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  type fields. ({issue}`23451`)
+* Fix writing of bloom filter columns in Parquet files. ({issue}`22701`)
+
+## Oracle connector
+
+* Improve performance for queries casting `char` or `varchar` columns. ({issue}`22728`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-460.md:44]: Is this for casts between char <> varchar or something else?

File: docs/src/main/sphinx/release/release-460.md (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-460.md:9]: Seems like all queries that need column info from looking at the PR so the generic text is correct.

File: docs/src/main/sphinx/release/release-460.md (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-460.md:19]: I think it relates to some incompatibility with dictionaries that are larger than one page. Not sure how to phrase that so users understand or its vague enough

File: docs/src/main/sphinx/release/release-460.md (hunk 7 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-460.md:44]: Original suggestion was

```
# Oracle
* Add support for cast project pushdown with Char and Varchar data type. ({issue}`22728`
```

and I think its any cast of char or varchar. I reworded ..

File: docs/src/main/sphinx/release/release-460.md (hunk 5 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-460.md:19]: What was the failure mode? Missing bloom filters? Broken parquet files? Query failures?

Comment (mosabua) [reply docs/src/main/sphinx/release/release-460.md:19]: Alter table execute optimize failed from the reported issue, but I think there might have been others as well. Maybe

Prevent query failures when writing Parquet files with Bloom filter on columns.

But I am honestly not sure that captures it correctly as well.

File: docs/src/main/sphinx/release/release-460.md (hunk 7 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-460.md:44]: I think the rewording I did is correct.

File: docs/src/main/sphinx/release/release-460.md (hunk 5 above)
Comment (raunaqmorarka) [reply docs/src/main/sphinx/release/release-460.md:19]: Any insert query could fail when the writer for a column configured with a bloom filter would fall back from dictionary encoding to plain encoding due to large cardinality.
We didn't produce broken files as the failure prevents the insert from completing.

File: docs/src/main/sphinx/release/release-460.md (hunk 7 above)
Comment (krvikash) [reply docs/src/main/sphinx/release/release-460.md:44]: > Is this for casts between char <> varchar or something else?

supporting cast pushdown when casting to `char` or `varchar`

```
char(m) -> char(n)
varchar(m) -> char(n)
```

```
varchar(m) -> varchar(n)
tinyint -> varchar(n)
smallint -> varchar(n)
int -> varchar(n)
bigint -> varchar(n)
decimal(p, q) -> varchar(n)
```

Comment (mosabua) [reply docs/src/main/sphinx/release/release-460.md:44]: oh.. the important thing is casting TO char or varchar.. will update

File: docs/src/main/sphinx/release/release-460.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
+# Release 460 (3 Oct 2024)
+
+## General
+
+* Fix query failures when complex symbols are referenced from within lambda expression. ({issue}`23649`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-460.md:5]: This needs to be reworded. "complex symbols" isn't something users understand. Maybe just say:

"Fix failure for certain queries involving lambda expressions".

File: docs/src/main/sphinx/release/release-460.md (hunk 5 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-460.md]: Or simply, "Fix query failures when writing bloom filters in Parquet files"

File: docs/src/main/sphinx/release/release-460.md (hunk 8 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-460.md:5]: Done

File: docs/src/main/sphinx/release/release-460.md (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-460.md]: Done

**Pull request merged**